server/app/views/components/SelectWithLabel.java

```diff
@@ -93,21 +102,49 @@ public DivTag getSelectTag() {
       customOptions.forEach(option -> fieldTag.with(option));
     } else {
       fieldTag.with(
-          options.stream()
+          optionGroups.stream()
               .map(
-                  optionData -> {
-                    return option(optionData.label())
-                        .withClasses(
-                            ReferenceClasses.MULTI_OPTION_QUESTION_OPTION,
-                            ReferenceClasses.MULTI_OPTION_VALUE)
-                        .withValue(optionData.value())
-                        .withCondSelected(optionData.value().equals(fieldValue));
+                  optionGroup -> {
+                    return optgroup()
+                        .withLabel(optionGroup.label())
+                        .with(
+                            each(
+                                optionGroup.options(),
+                                optionData -> {
+                                  return option(optionData.label())
```

Comment:
Consider pulling this each computation out for readability, it'd remove a lot of nesting and a lambda return.

Reply:
I moved the computation of both the `<optgroup>` and `<option>` tags out into separate methods, which removed a fair amount of the nesting, which is what I think you're after.